shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1679,37 +1755,45 @@ - (UITextPosition*)closestPositionToPoint:(CGPoint)point withinRange:(UITextRang
   NSUInteger start = ((FlutterTextPosition*)range.start).index;
   NSUInteger end = ((FlutterTextPosition*)range.end).index;
 
-  NSUInteger _closestIndex = 0;
-  CGRect _closestRect = CGRectZero;
-  NSUInteger _closestPosition = 0;
+  // Selecting text using the floating cursor is not as precise as the pencil.
+  // Allow further vertical deviation and base more of the decision on horizontal comparison.
+  CGFloat verticalPrecision = _isFloatingCursorActive ? 10 : 1;
+
+  BOOL isFirst = YES;
+  NSUInteger _closestRectIndex = 0;
   for (NSUInteger i = 0; i < [_selectionRects count]; i++) {
     NSUInteger position = _selectionRects[i].position;
     if (position >= start && position <= end) {
```

Comment:
probably not related to your PR, but this check is odd - when will `position` ever be out of `[start, end]` range?

Reply:
Maybe UIKit will call `closestPositionToPoint:withinRange:` where `range` is limited.

Reply:
ic. the selectionRects contains all the rects, which may fall outside of the given range.